oauth_mgmt/utils_test.py

```diff
@@ -0,0 +1,113 @@
+"""
+Util tests.
+"""
+# pylint: disable=no-self-use,unused-argument
+from datetime import timedelta
+import json
+import mock
+
+from django.test import TestCase
+from django.utils.timezone import now
+
+import pytest
+
+from .factories import BackingInstanceFactory, PrefetchBackingInstanceFactory
+from .models import BackingInstance
+from .utils import get_access_token, UnretrievableToken
+
+
+class GetAccessTokenTests(TestCase):
+    """
+    Tests for get_access_token
+    """
+    def test_current_token_returns(self):
+        """
+        If the token is current, return it.
+        """
+        bi = BackingInstanceFactory.create(
+            instance_url='http://valid',
+            access_token='asdf',
+            access_token_expiration=now() + timedelta(days=1))
+
+        result = get_access_token(bi)
+
+        assert result == 'asdf'
+
+    def test_fetch_errors_throws_exception(self):
+        """
+        If the access token fetch returns non-200, throw exception.
+        """
+        bi = PrefetchBackingInstanceFactory.create(
+            instance_url='http://no_access',
+            grant_token='qwerty')
+        assert bi.is_expired
+
+        with mock.patch('oauth_mgmt.utils.requests', autospec=True) as m_req:
+            m_req.post.return_value.status_code = 500
+            with pytest.raises(UnretrievableToken):
+                get_access_token(bi)
+
+    def test_grant_fetch_updates_db(self):
+        """
+        Info saved to db on grant fetch.
+        """
+        bi = PrefetchBackingInstanceFactory.create(
+            instance_url='http://no_access',
+            grant_token='qwerty')
+
+        with mock.patch('oauth_mgmt.utils.requests', autospec=True) as m_req:
+            m_req.post.return_value.content = """{
```

Comment:
Is there a way to assert that the data being sent to the request is correct? At least the grant type?